### PR TITLE
fix 12 hour format logic & write supporting test cases

### PR DESCRIPTION
**Pull request by nijilnarayanan**

### CODE REVIEW

File: samay.js [hunk 1]
```diff
@@ -2,6 +2,32 @@
 
 function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
 
+/* global window */
+
+function calcDate(structuredDate) {
+
+	try {
+		var splitDate = structuredDate.split('T'),
+		    datePart = splitDate[0].split('-'),
```
Comment (jigar1101) [inline samay.js:11]: Might want to check if the date exists?

File: samay.js [hunk 2]
```diff
@@ -215,7 +255,7 @@ function _format(format) {
 	fullYear = inputDate.getFullYear(),
 	    year = fullYear % 100,
 	    hour24 = inputDate.getHours(),
-	    hour12 = hour24 > 12 ? hour24 % 12 : hour24,
+	    hour12 = hour24 % 12 + (hour24 / 12 == 0 || hour24 / 12 == 1 ? 12 : 0),
```
Comment (jigar1101) [inline samay.js]: `===` instead of `==` ?

**Review by jigar1101 — COMMENTED** [review]
@nijilnarayanan Great work 💯

File: samay.js (hunk 2 above)
Comment (nijilnarayanan) [reply samay.js]: Done

File: samay.js (hunk 1 above)
Comment (nijilnarayanan) [reply samay.js:11]: Enclosed in try catch. As per @vihangpatel

**Review by vihangpatel — APPROVED** [review]

**Pull request merged**